### PR TITLE
Print task trees

### PR DESCRIPTION
**Pull request by tomas-nava**
### Description
Make it possible to print task trees from the rails console.

```
> puts RootTask.last.structure_render(:id, :status, :closed_at)

RootTask 220, on_hold, (closed_at)
├── BvaDispatchTask 226, on_hold, (closed_at)
│   └── BvaDispatchTask 227, assigned, (closed_at)
├── JudgeAssignTask 224, completed, 2019-06-14 22:35:44 UTC
│   └── AttorneyTask 225, completed, 2019-06-14 22:35:44 UTC
└── DistributionTask 221, on_hold, (closed_at)
    └── HearingTask 222, on_hold, (closed_at)
        └── ScheduleHearingTask 223, assigned, (closed_at)
```

```
> puts LegacyAppeal.first.structure_render(:id, :status, :created_at)

LegacyAppeal 1
├── RootTask 347, assigned, 2019-06-14 22:35:49 UTC
└── ColocatedTask 348, on_hold, 2019-06-14 22:35:49 UTC
    └── ColocatedTask 349, on_hold, 2019-06-14 22:35:49 UTC
```

### CODE REVIEW

**Comment by lowellrex** [comment]
I'm wondering if we can modify the interface for this slightly to accomodate `LegacyAppeals` with more than one branch of tasks (https://caseflow-looker.va.gov/sql/dbf5rgsnkhbbvk) by accepting an appeal. Perhaps a wrapper that looks something like:
```ruby
def pretty_print_tasks_for(appeal, atts)
  tasks = appeal.tasks.where(parent_id: nil)
  TTY::Tree.new({ "#{appeal.id}": tasks.map { |child| child.structure(false, *atts) } }).render
end
```

File: app/models/task.rb [hunk 1]
```diff
@@ -49,6 +49,19 @@ class << self; undef_method :open; end
...
+    if children.count.zero?
+      if base || parent.nil?
+        { "#{leaf_name}": [] }
+      else
+        leaf_name
+      end
+    else
+      { "#{leaf_name}": children.map { |child| child.structure(false, *atts) } }
+    end
+  end
```
Comment (lowellrex) [inline app/models/task.rb]: In my brief test drive with `tty-tree`, I think the gem correctly displays empty arrays so we might be able simplify this method to:
```ruby
def structure(*atts)
  leaf_name = "#{self.class.name} #{attributes_to_s(*atts)}"

  { "#{leaf_name}": children.map { |child| child.structure(*atts) } }
end
```

```ruby
rails c> puts TTY::Tree.new(MailTask.last.appeal.root_task.structure(:id, :status, :closed_at)).render
RootTask 435, on_hold, (closed_at)
├── CongressionalInterestMailTask 446, on_hold, (closed_at)
│   └── CongressionalInterestMailTask 447, in_progress, (closed_at)
├── AddressChangeMailTask 443, cancelled, 2019-06-05 17:54:35 UTC
│   └── AddressChangeMailTask 444, cancelled, 2019-06-05 17:54:35 UTC
│       └── AddressChangeMailTask 445, cancelled, 2019-06-05 17:54:35 UTC
├── CongressionalInterestMailTask 441, cancelled, 2019-06-05 17:53:31 UTC
│   └── CongressionalInterestMailTask 442, cancelled, 2019-06-05 17:53:31 UTC
├── DeathCertificateMailTask 438, cancelled, 2019-06-05 17:52:32 UTC
│   └── DeathCertificateMailTask 439, cancelled, 2019-06-05 17:52:32 UTC
│       └── DeathCertificateMailTask 440, cancelled, 2019-06-05 17:52:32 UTC
└── DistributionTask 436, on_hold, (closed_at)
    └── EvidenceSubmissionWindowTask 437, assigned, (closed_at)
```

Comment (tomas-nava) [reply app/models/task.rb]: just discovered this myself as well! 🥳

Comment (tomas-nava) [reply app/models/task.rb]: fixed in [this commit](https://github.com/department-of-veterans-affairs/caseflow/pull/11171/commits/0f896544e6e1fdeeb7a6fa0060e765c4e3d1c6ea)

**Comment by tomas-nava** [comment]
> I'm wondering if we can modify the interface for this slightly to accomodate `LegacyAppeals` with more than one branch of tasks (https://caseflow-looker.va.gov/sql/dbf5rgsnkhbbvk) by accepting an appeal.

Added methods to `Appeal` and `LegacyAppeal` in [this commit](https://github.com/department-of-veterans-affairs/caseflow/pull/11171/commits/e5933c1d151c124f10aa8cab82bdde13574dd8c3).

File: app/models/concerns/prints_task_tree.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    { "#{leaf_name}": task_tree_children.map { |child| child.structure(*atts) } }
+  end
+
+  private
+
+  def task_tree_children
+    return children.order(:id) if is_a? Task
+
+    tasks.where(parent_id: nil).order(:id)
+  end
```
Comment (lowellrex) [inline app/models/concerns/prints_task_tree.rb]: Clever! Really like this!

**Comment by lowellrex** [comment]
🎉

**Pull request merged**